### PR TITLE
Feature/copy paste from histogram #741

### PR DESCRIPTION
**Pull request by sol695510**
### Description of the Change

Added functionality to copy property values from a bar selection on the Histogram View to the clipboard via a right-click context menu or via ctrl+c. The context menu also provides an option to copy property values and their associated counts.

Also limited selections of bars to only left-click on the mouse instead of previously being activated by left-click, right-click or clicking the mouse wheel.

### Alternate Designs

N/A

### Why Should This Be In Core?

N/A

### Benefits

Copying values from the Histogram View will be more time efficient for users.

### Possible Drawbacks

N/A

### Verification Process

Tested by copying various bar selections, including both with shift held down or control held down, then either using ctrl+c or the right-click context menu to copy values to a text editor to verify whether the correct values were copied across.

### Applicable Issues

#741

### CODE REVIEW

**Comment by sol695510** [comment]
@antares1470 what's your call on the Code Smells detected? In my opinion the order of "if" statements provides better readability of the logic.

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:117]: final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:123]: final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:547]: merge this if statement with the above

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:557]: final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:556]: merge if statement with above

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:579]: final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:583]: final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:594]: merge if statement with above

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:628]: merge all these if statements together

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Comment by sol695510** [comment]
@antares1470 Why is the SonarCloud Code Analysis failing?

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:544]: Minor but I think we prefer `&&` at the start of the newline rather than the end of the old line

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:552]: same here

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:569]: same here

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:583]: same here

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:614]: same here

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Comment by antares1470** [comment]
> 
> 
> @antares1470 Why is the SonarCloud Code Analysis failing?

My guess would be the code coverage but I'm not entirely sure. I wouldn't worry about it too much to be honest. You've cleared all the code smells which is fine

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (formalhaut69) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:524]: Also can be final

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java
Comment (formalhaut69) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramDisplay.java:522]: A little nitpicky but includeCounts can be final

**Review by arcturus2 — APPROVED** [review]
Nice work @sol695510

**Pull request merged**
